### PR TITLE
feat: support no internet connection offline local

### PR DESCRIPTION
**Pull request by TzlilSwimmer123**

### CODE REVIEW

File: pkg/networkValidator/networkValidator.go [hunk 1]
```diff
@@ -21,7 +22,7 @@ func (nv *NetworkValidator) SetOfflineMode(offlineMode string) {
 }
 
 func (nv *NetworkValidator) IdentifyNetworkError(errStr string) error {
-	if errStr == "network error" {
+	if errStr == "network error" || strings.Contains(errStr, "connection refused") || strings.Contains(errStr, "no such host") {
 		if nv.offlineMode == "fail" {
```
Comment (TzlilSwimmer123) [inline pkg/networkValidator/networkValidator.go:26]: We can't really cover 100% of network errors due to network components react to no internet / backend is down. ATM we focus on adding more possible errors

File: bl/validation/k8sValidator.go [hunk 2]
```diff
@@ -154,7 +154,7 @@ func (val *K8sValidator) validateResource(filepath string) (bool, []error, strin
 }
 
 func (val *K8sValidator) isNetworkError(errorString string) bool {
-	return strings.Contains(errorString, "no such host")
+	return strings.Contains(errorString, "no such host") || strings.Contains(errorString, "connection refused")
```
Comment (TzlilSwimmer123) [inline bl/validation/k8sValidator.go:157]: We can't really cover 100% of network errors due to network components react to no internet / backend is down. ATM we focus on adding more possible errors

File: cmd/config/main.go [hunk 3]
```diff
@@ -53,7 +55,10 @@ func New(ctx *ConfigCommandContext) *cobra.Command {
 
 func validateKey(key string) error {
 	validKeys := make(map[string]bool)
-	validKeys["token"] = true
+
+	for _, key := range ConfigAvailableKeys {
+		validKeys[key] = true
+	}
```
Comment (TzlilSwimmer123) [inline cmd/config/main.go:61]: In todays build we missed the possibility for config set offline!
This change suppose to couple between adding new tokens and make them valid!

File: pkg/networkValidator/validator_test.go [hunk 4]
```diff
@@ -41,6 +51,11 @@ func test_identifyNetworkError_network_error(validator *NetworkValidator, offlin
 	return validator.IdentifyNetworkError("network error")
 }
 
+func test_identifyNoInternet_network_error(validator *NetworkValidator, offlineMode string) error {
+	validator.SetOfflineMode(offlineMode)
+	return validator.IdentifyNetworkError("no such host")
```
Comment (alexfedin) [inline pkg/networkValidator/validator_test.go:66]: it's worth adding tests for the other errors that indicate a network error

**Review by alexfedin — DISMISSED** [review]

File: pkg/networkValidator/validator_test.go (hunk 4 above)
Comment (TzlilSwimmer123) [reply pkg/networkValidator/validator_test.go:66]: We have 2/3. I thought it's not the important to add the connection refused one. Just added it

**Review by dimabru — APPROVED** [review]

**Pull request merged**